### PR TITLE
Worker Indexing - Adding log to inform about mixed function app

### LINKED ISSUE #8110: Worker Indexing - Adding log to inform about mixed function app
Detecting mixed app and adding log message to inform if customer's function app is in mixed format (some functions following legacy app format while others following new format for worker indexing). This log message will later on assist us in writing detectors to help with CRIs on mixed app.

### PR DESCRIPTION
**Pull request by surgupta-msft**
resolves #8110 

Detecting mixed app and adding log message to inform if customer's function app is in mixed format (some functions following legacy app format while others following new format for worker indexing). This log message will later on assist us in writing detectors to help with CRIs on mixed app.

### CODE REVIEW

File: src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs [hunk 1]
```diff
@@ -23,17 +25,20 @@ public class AggregateFunctionMetadataProvider : IFunctionMetadataProvider
         private readonly Dictionary<string, ICollection<string>> _functionErrors = new Dictionary<string, ICollection<string>>();
         private readonly ILogger _logger;
         private readonly IFunctionInvocationDispatcher _dispatcher;
+        private readonly IOptionsMonitor<ScriptApplicationHostOptions> _applicationHostOptions;
         private ImmutableArray<FunctionMetadata> _functions;
         private IFunctionMetadataProvider _hostFunctionMetadataProvider;
 
         public AggregateFunctionMetadataProvider(
-            ILogger<AggregateFunctionMetadataProvider> logger,
```
Comment (soninaren) [inline src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs]: Why remove this?

File: src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs [hunk 2]
```diff
@@ -81,6 +89,30 @@ public async Task<ImmutableArray<FunctionMetadata>> GetFunctionMetadataAsync(IEn
...
+                    if (Utility.TryReadFunctionConfig(functionDirectory, out string json, fileSystem))
+                    {
+                        mixedApp = true;
+                        break;
+                    }
+                }
+
+                if (mixedApp)
+                {
+                    logger.Log(LogLevel.Information, "Detected mixed function app. All functions may not be indexed.");
```
Comment (soninaren) [inline src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs]: Would be useful to get the list of all functions identified to be in legacy format. This way if the customer says why was this not loaded we can validate against the list.

File: src/WebJobs.Script/Host/FunctionMetadataManager.cs [hunk 3]
```diff
@@ -36,13 +37,14 @@ public class FunctionMetadataManager : IFunctionMetadataManager
         private ConcurrentDictionary<string, FunctionMetadata> _functionMetadataMap = new ConcurrentDictionary<string, FunctionMetadata>(StringComparer.OrdinalIgnoreCase);
 
         public FunctionMetadataManager(IOptions<ScriptJobHostOptions> scriptOptions, IFunctionMetadataProvider functionMetadataProvider,
-            IOptions<HttpWorkerOptions> httpWorkerOptions, IScriptHostManager scriptHostManager, ILoggerFactory loggerFactory, IOptions<LanguageWorkerOptions> languageWorkerOptions, IEnvironment environment)
+            IOptions<HttpWorkerOptions> httpWorkerOptions, IScriptHostManager scriptHostManager, ILoggerFactory loggerFactory,
+            IOptions<LanguageWorkerOptions> languageWorkerOptions, IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IEnvironment environment)
         {
             _scriptOptions = scriptOptions;
```
Comment (soninaren) [inline src/WebJobs.Script/Host/FunctionMetadataManager.cs:42]: You may be able to get script path from here.

**Review by satvu — APPROVED** [review]

**Review by soninaren — CHANGES_REQUESTED** [review]

File: src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs (hunk 1 above)
Comment (surgupta-msft) [reply src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs]: Discussed in a different PR

File: src/WebJobs.Script/Host/FunctionMetadataManager.cs (hunk 3 above)
Comment (surgupta-msft) [reply src/WebJobs.Script/Host/FunctionMetadataManager.cs:42]: Done

File: src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs (hunk 2 above)
Comment (surgupta-msft) [reply src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs]: Updated code here.

**Review by soninaren — APPROVED** [review]

**Pull request merged**